### PR TITLE
:seedling:  Make LogicalCluster claimable

### PR DESCRIPTION
**Pull request by mjudeikis**
## Summary

Attempt to make LogicalCluster native resource so it can be used in claims. 

## Related issue(s)

Fixes #

## Release Notes

```release-note
Make LogicalCluster claimable resource for deeper level integration ontop of KCP
```

### CODE REVIEW

**Comment by mjudeikis** [comment]
Not working. Checking

File: test/e2e/virtual/apiexport/virtualworkspace_test.go [hunk 1]
```diff
@@ -825,8 +825,8 @@ func gatherInternalAPIs(t *testing.T, discoveryClient discovery.DiscoveryInterfa
 	for _, apiResourcesList := range apiResourcesLists {
 		gv, err := schema.ParseGroupVersion(apiResourcesList.GroupVersion)
 		require.NoError(t, err)
-		// ignore kcp resources
-		if strings.HasSuffix(gv.Group, ".kcp.io") {
+		// ignore kcp resources expect for core.kcp.io
```
Comment (sttts) [inline test/e2e/virtual/apiexport/virtualworkspace_test.go:828]: grammar

**Comment by sttts** [comment]
A nit. Otherwise lgtm.

**Comment by sttts** [comment]
/approve
/lgtm
/hold

for the nit.

**Comment by mjudeikis** [comment]
/retest 
different flakes :/

**Pull request merged**